google/cloud/opentelemetry/internal/monitored_resource_test.cc

```diff
@@ -27,6 +27,46 @@ namespace sc = opentelemetry::sdk::resource::SemanticConventions;
 using ::testing::Pair;
 using ::testing::UnorderedElementsAre;
 
+TEST(AsString, Simple) {
+  struct TestCase {
+    opentelemetry::sdk::common::OwnedAttributeValue value;
+    std::string result;
+  };
+  std::vector<TestCase> cases = {
+      {true, "true"},       {false, "false"},
+      {int32_t{1}, "1"},    {uint32_t{2}, "2"},
+      {int64_t{3}, "3"},    {uint64_t{4}, "4"},
+      {double{5.6}, "5.6"}, {std::string{"value"}, "value"},
```

Comment:
Perhaps we should quote string attributes to distinguish the output of `true` and `"true"`.

Reply:
I considered that, but I don't think preserving the type is a feature. The majority of these are going to be strings. And in other places we expect the resulting string to not be quoted.

Reply:
OK.